src/shared/loop-util.c

```diff
@@ -178,11 +179,19 @@ LoopDevice* loop_device_unref(LoopDevice *d) {
 
                 control = open("/dev/loop-control", O_RDWR|O_CLOEXEC|O_NOCTTY|O_NONBLOCK);
                 if (control < 0)
-                        log_debug_errno(errno, "Failed to open loop control device: %m");
-                else {
-                        if (ioctl(control, LOOP_CTL_REMOVE, d->nr) < 0)
-                                log_debug_errno(errno, "Failed to remove loop device: %m");
-                }
+                        log_warning_errno(errno,
+                                          "Failed to open loop control device, cannot remove loop device %s: %m",
+                                          strna(d->node));
+                else
+                        for (unsigned n_attempts = 0;;) {
+                                if (ioctl(control, LOOP_CTL_REMOVE, d->nr) >= 0)
+                                        break;
+                                if (errno != EBUSY || ++n_attempts >= 64) {
+                                        log_warning_errno(errno, "Failed to remove device %s: %m", strna(d->node));
+                                        break;
+                                }
+                                usleep(50 * USEC_PER_MSEC);
+                        }
```

Comment:
urks, but ok...

given allocated devices are reused too by our allocation logic my prior thinking was to just ignore the issue: whether the loopback device exists as node or not doesn't matter too much...

btw, i just realized a race here: when this stuff runs and some other code tries to allocate a loopback device, then it might happen the kernel tells us via LOOP_CTL_GET_FREE that some loopback device is free, and then we try to open it but someone else might already have used it, released it again, and used LOOP_CTL_REMOVE on it, so that our attempt to open it will fail with ENOENT. We need to tweak the allocation loop hence to retry to open() failing with ENOENT, too.